CMakeLists.txt

```diff
@@ -1052,14 +1053,21 @@ if (NOT ANDROID OR TERMUX)
     install(DIRECTORY doc DESTINATION ${CMAKE_INSTALL_DOCDIR})
     install(DIRECTORY papers DESTINATION ${CMAKE_INSTALL_DOCDIR}/other)
     install(FILES LICENSE DESTINATION ${CMAKE_INSTALL_DATADIR}/licenses/furnace)
-    if (WITH_DEMOS)
-      install(DIRECTORY demos DESTINATION ${CMAKE_INSTALL_DATADIR}/furnace)
-    endif()
-    if (WITH_INSTRUMENTS)
-      install(DIRECTORY instruments DESTINATION ${CMAKE_INSTALL_DATADIR}/furnace)
-    endif()
-    if (WITH_WAVETABLES)
-      install(DIRECTORY wavetables DESTINATION ${CMAKE_INSTALL_DATADIR}/furnace)
+    if (WITH_DEMOS OR WITH_INSTRUMENTS OR WITH_WAVETABLES)
+      set(FURNACE_DATADIR ${CMAKE_INSTALL_DATADIR}/furnace)
+      if (WITH_DEMOS)
+        install(DIRECTORY demos DESTINATION ${FURNACE_DATADIR})
+      endif()
+      if (WITH_INSTRUMENTS)
+        install(DIRECTORY instruments DESTINATION ${FURNACE_DATADIR})
+      endif()
+      if (WITH_WAVETABLES)
+        install(DIRECTORY wavetables DESTINATION ${FURNACE_DATADIR})
+      endif()
+      add_compile_definitions(FURNACE_DATADIR="${CMAKE_INSTALL_PREFIX}/${FURNACE_DATADIR}")
```

Comment:
This is a wrong assumption: `CMAKE_INSTALL_DATADIR` (as used in `FURNACE_DATADIR`) may be
- an absolute path that is
- not within `CMAKE_INSTALL_PREFIX`

Please use either
- `CMAKE_INSTALL_FULL_DATADIR` for a guaranteed-absolute location,
- `cmake_path` to join `CMAKE_INSTALL_PREFIX` and `FURNACE_DATADIR` when possible (requires CMake 3.20 or higher), or
- The [`JoinPaths`](https://github.com/jtojnar/cmake-snips) CMake module for an all-version variant of `cmake_path`